pkgs/build-support/fetchnodemodules/default.nix

```diff
@@ -0,0 +1,102 @@
+{ lib, stdenvNoCC, makeWrapper, nodejs }:
+
+{ src
+, hash ? ""
+, sha256 ? ""
+, name ? null
+, production ? true
+, runScripts ? false
+, makeTarball ? true
+, preferLocalBuild ? true
+, exposeBin ? false
+, npmFlags ? ""
+, ...
+} @ args:
+
+assert makeTarball -> !exposeBin;
+let
+  suffix = if makeTarball then ".tar.gz" else "";
+  name_ = if name == null then "${src.name}-node_modules${suffix}" else name;
+  nodejs_ = if args ? nodejs then args.nodejs else nodejs;
+
+  hash_ =
+    if hash != "" then { outputHashAlgo = null; outputHash = hash; }
+    else if sha256 != "" then { outputHashAlgo = "sha256"; outputHash = sha256; }
+    else throw "fetchNodeModules requires a hash for ${name_}";
+
+  customArgs = [ "name" "hash" "sha256" ];
+in stdenvNoCC.mkDerivation ({
+  inherit src production runScripts makeTarball preferLocalBuild;
+
+  name = name_;
+  nativeBuildInputs = [ nodejs_ makeWrapper ];
+
+  phases = "unpackPhase patchPhase buildPhase installPhase";
```

Comment:
just noticed, probably not important at all, but the phases attribute is being (already?) deprecated.

Reply:
I wasn't aware of this. If anyone else wants to follow up on this, [this](https://github.com/NixOS/nixpkgs/issues/28910) looks like the related issue.